### PR TITLE
feat(e2esinktest): switch to stream_chunk

### PR DESCRIPTION
**Pull request by idx0-dev**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Since additional data types have been supported in our Java binding (#8740) and certain encoding behaviors in the JSON payload will be align with more universal methods (I could do it only if the `e2e-sink-test` does not depend on it), I have switched the payload format from JSON to stream_chunk and fixed some test data. ( It seems that the original test data was copied and pasted from the output, and has a slight issue.)

I will cleanup 'RW_CONNECTOR_RPC_SINK_PAYLOAD_FORMAT' env var later.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]
LGTM

**Comment by fuyufjh** [comment]
By the way, I would strongly recommend to substitute the JNI implementation (`com.risingwave.java.binding.Binding`) for a pure Java implementation. Have once discussed with @wenym1 .

**Comment by wenym1** [comment]
> By the way, I would strongly recommend to substitute the JNI implementation (com.risingwave.java.binding.Binding) of serialization/deserialization for a pure Java implementation.

I think before we decide to substitute the implementation we may need some benchmark to see if the cost of JNI call is significant. @idx0-dev Could you make a testing on it?

File: e2e_test/sink/remote/jdbc.check.pg.slt [hunk 1]
```diff
@@ -3,8 +3,8 @@
 query I
 select * from t_remote order by id;
 ----
-1 Alex 28208 281620391 4986480304337356800 28162.0391 2.03 28162.0391 2023-03-20 10:18:30
```
Comment (wenym1) [inline e2e_test/sink/remote/jdbc.check.pg.slt:6]: It is because we are using stream chunk that the result gets more precise?

**Review by wenym1 — APPROVED** [review]
LGTM

**Review by tabVersion — APPROVED** [review]

File: e2e_test/sink/remote/jdbc.check.pg.slt (hunk 1 above)
Comment (idx0-dev) [reply e2e_test/sink/remote/jdbc.check.pg.slt:6]: No. The origin test data is wrong. It should not lost precision anyway.

**Comment by idx0-dev** [comment]
> > By the way, I would strongly recommend to substitute the JNI implementation (com.risingwave.java.binding.Binding) of serialization/deserialization for a pure Java implementation.
> 
> I think before we decide to substitute the implementation we may need some benchmark to see if the cost of JNI call is significant. @idx0-dev Could you make a testing on it?

I will do it later. #8942

**Pull request merged**